pkg/prometheus/promcfg_test.go

```diff
@@ -8980,6 +8980,30 @@ scrape_configs:
 scrape_configs:
 - job_name: scrapeconfig/default/testscrapeconfig1
   scheme: https
+`,
+		},
+		{
```

Comment:
can you add 2 other tests?
* 1 with the enforced* limits + scrapeconfig limits being defined
* 1 with only the enforced* limits being defined

Reply:
There are `TestSampleLimits`, `TestTargetLimits`, `TestLabelLimits`, `TestLabelNameLengthLimits` and `TestLabelValueLengthLimits` ([example](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg_test.go#L6319-L6535)) tests that already testing same functionality and many other behaviours of `AddLimitToTAML` function which reused in that PR.

Not sure if that needs to be duplicated specifically for ScrapeConfig.

Reply:
hmm ok, I think that we can extend the existing unit tests to cover all combinations (pod monitors, service monitors, probes and scrapeconfigs) but it doesn't have to be in this PR.